evmcore/validation_test.go

```diff
@@ -0,0 +1,774 @@
+package evmcore
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/kzg4844"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateTx_RejectsNonLegacyTransactions_BeforeEip2718(t *testing.T) {
+	for name, tx := range getTxsFromAllTypes() {
+		if _, ok := tx.(*types.LegacyTx); ok {
+			continue // Skip legacy transactions
+		}
+		t.Run(name, func(t *testing.T) {
+			err := validateTx(types.NewTx(tx), types.NewPragueSigner(big.NewInt(1)),
+				validationOptions{eip2718: false})
+			require.ErrorIs(t, ErrTxTypeNotSupported, err)
+		})
+	}
+}
+
+// testTransactionsOption is a set of options to adjust the validation of transactions
+// so that it would accept all types of transactions, considering them as local transactions
+// with a min tip of 1, current base fee of 1, and a current max gas of 100_000.
+func testTransactionsOption() validationOptions {
```

Comment:
I'd say either
```suggestion
var testTransactionsOption = validationOptions {
```
or
```suggestion
func getTestTransactionsOption() validationOptions {
```
The function option has the advantage that the there can not be an accidental modification due to sharing the same instance among multiple tests.

Reply:
I like the function option better. Will rename accordingly.

Reply:
https://github.com/0xsoniclabs/sonic/pull/154/commits/4205286062d9a53c041a02d2ccb9f0719ab7ba0f